agent/agent.go

```diff
@@ -373,9 +364,26 @@ func runChecker(ctx context.Context, checker health.Checker, probeCh chan<- heal
 
 	log.Debugf("Running checker %q.", checker.Name())
 
-	var probes health.Probes
-	checker.Check(ctx, &probes)
-	probeCh <- probes
+	checkResult := make(chan health.Probes)
```

Comment:
It's safer to set chan capacity to 1 - otherwise it might block and the goroutine will be left behind.
Also it's simpler when all dependent goroutines wrap up when interrupted so we don't have to wonder if anything's stuck.